application/src/main/java/org/togetherjava/tjbot/commands/help/HelpSystemHelper.java

```diff
@@ -363,19 +363,19 @@ private void executeNoActivityAdviceCheck(long threadChannelId, long authorId) {
                     threadChannelId, authorId);
             return;
         }
-        if (noAuthorActivity(authorId, threadChannel)) {
+        if (hasNoAuthorActivity(authorId, threadChannel)) {
             MessageEmbed embed = HelpSystemHelper.embedWith(
                     """
-                            Hey there %s👋  It has been a bit after you created this thread and you still did not share any details of your question.
+                            Hey there %s👋 It has been a bit after you created this thread and you still did not share any details of your question.
                             Helpers have seen your question already and are just waiting for you to elaborate on your problem and provide detailed information on it 👌
                             """
                         .formatted(User.fromId(authorId).getAsMention()));
             threadChannel.sendMessageEmbeds(embed).queue();
         }
     }
 
-    private static boolean noAuthorActivity(long authorId, ThreadChannel threadChannel) {
-        return threadChannel.getIterableHistory()
+    private static boolean hasNoAuthorActivity(long authorId, ThreadChannel threadChannel) {
+        return threadChannel.getHistory().size() >= 10 || threadChannel.getIterableHistory()
```

Comment:
1. now ur sending two requests to discord
2. instead of using iterable history, might want to use `getHistory(10)`.
3. might be wrong, but iterable history is blocking, no? u need a rest action here

Reply:
@Tais993 for clarification

Reply:
This doesn't work for multiple reasons

getHistory returns an empty history, so history will always be 0

iterablehistory requires you to retrieve messages, I very much doubt the stream will have any elements


So
1, incorrect, 0 is the answer
2. getHistory(10) doesn't exist
3. getIterableHistory is the only one that supports good async retrieval, the others also allow async, but aren't near the quality of getIterableHistory


Has this been tested properly? Does this work? As I don't think it works after reading this.
In all fairness history is complicated I'd say, so I can also be mistaken.